src/serde_util.rs

```diff
@@ -58,6 +58,27 @@ where
     Ok(millis.map(Duration::from_millis))
 }
 
+pub(crate) fn deserialize_cursor_type<'de, D>(
```

Comment:
This implementation should be equivalent to adding the following tags to the cursor type:
```
#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum CursorType { ... }
```
and adding the following tag to the `cursor_type` field on the options:
```
#[serde(default)]
pub cursor_type: Option<CursorType>,
```
`default` here is necessary to deserialize into an optional type properly: if there is a value specified, then the `Deserialize` implementation for `CursorType` will be used, and if not, the deserializer will fall back to the default value of `None`.